### PR TITLE
Add an option to use Docker instead of Podman

### PR DESCRIPTION
**Pull request by maelk**
Creates the environment variable CONTAINER_RUNTIME . The variable can be set to "docker" or "podman" depending on the runtime you want to use. It defaults to Podman. The requested runtime will be installed automatically but might conflict if a different runtime was previously installed

### CODE REVIEW

File: ubuntu_install_requirements.sh [hunk 1]
```diff
@@ -97,7 +98,26 @@ sudo apt -y install \
...
+    software-properties-common
+  curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
+  sudo add-apt-repository \
+    "deb [arch=amd64] https://download.docker.com/linux/ubuntu \
+    $(lsb_release -cs) \
+    stable"
+  sudo apt update
+  sudo apt install -y docker-ce docker-ce-cli containerd.io
+  sudo systemctl start docker
+fi
```
Comment (namnx228) [inline ubuntu_install_requirements.sh:120]: This duplicated change should be avoided. I'm also making a patch that both ubuntu and centos installation requirement scripts `source` another script that contains the stuffs which are the same.

File: centos_install_requirements.sh [hunk 2]
```diff
@@ -77,6 +77,19 @@ sudo yum -y install \
   unzip \
   genisoimage
 
+if [ "${CONTAINER_RUNTIME}" == "podman" ]; then
```
Comment (kanwar-saad) [inline centos_install_requirements.sh]: it should  be either 
[[ "${CONTAINER_RUNTIME}" == "podman" ]]
or
[ "${CONTAINER_RUNTIME}" = "podman" ]

File: ubuntu_install_requirements.sh (hunk 1 above)
Comment (maelk) [reply ubuntu_install_requirements.sh:120]: It is better to separate independent changes in different commits and pull requests to have the smallest changes. Here, adding a new file sourced for one shared line is not justified IMO. Removing duplication is a good idea, but should be done  for the whole Ubuntu and CentOS files in a next commit.

File: centos_install_requirements.sh (hunk 2 above)
Comment (maelk) [reply centos_install_requirements.sh]: Fixed in three files

File: lib/common.sh [hunk 3]
```diff
@@ -32,6 +32,8 @@ MANAGE_INT_BRIDGE=${MANAGE_INT_BRIDGE:-y}
 INT_IF=${INT_IF:-}
 #Root disk to deploy coreOS - use /dev/sda on BM
 ROOT_DISK_NAME=${ROOT_DISK_NAME-"/dev/sda"}
+#Container runtime
+CONTAINER_RUNTIME=${CONTAINER_RUNTIME:-"podman"}
```
Comment (kanwar-saad) [inline lib/common.sh:36]: Its a general comment.
${CONTAINER_RUNTIME:-"podman"}
can be changed to
"${CONTAINER_RUNTIME:-podman}"

File: host_cleanup.sh [hunk 4]
```diff
@@ -6,13 +6,13 @@ source lib/common.sh
 
 # Kill and remove the running ironic containers
 for name in ironic ironic-inspector dnsmasq httpd mariadb; do
-    sudo podman ps | grep -w "$name$" && sudo podman kill $name
-    sudo podman ps --all | grep -w "$name$" && sudo podman rm $name -f
+    sudo ${CONTAINER_RUNTIME} ps | grep -w "$name$" && sudo ${CONTAINER_RUNTIME} kill $name
```
Comment (kanwar-saad) [inline host_cleanup.sh:9]: quotes around variables.
same for other places.

**Review by kanwar-saad — DISMISSED** [review]

File: host_cleanup.sh (hunk 4 above)
Comment (maelk) [reply host_cleanup.sh:9]: leaving it as is for now, we will fix the style in the whole file afterwards

**Comment by maelk** [comment]
squashed and PR upstream : https://github.com/metal3-io/metal3-dev-env/pull/36

**Pull request merged**